ganttproject/src/net/sourceforge/ganttproject/TaskContainmentHierarchyFacadeImpl.java

```diff
@@ -136,6 +124,52 @@ public Task getContainer(Task nestedTask) {
   }
 
   @Override
+  public void sort(Comparator<Task> comparator) {
+    sortHelper(getRootTask(), comparator);
+  }
+
+  private void sortHelper(Task root, Comparator<Task> comparator) {
+    Task[] tasks = getNestedTasks(root);
+    // Bubble sort
+    for (int i = 0; i < tasks.length; i++) {
```

Comment:
If I were you I would just remove all nodes from parent, sort them using `Collections.sort` and add back to the parent in the sorted order. This will make the code more simple and less fragile.